### PR TITLE
Adjust TAPI cross-version test

### PR DESCRIPTION
**Pull request by donat**
- Move Gradle 8.2 test to the r82 package
- Add test for old behavior

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test rFN

File: subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r82/TestLauncherTestSpecCrossVersionSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+@ToolingApiVersion('>=7.6')
+@TargetGradleVersion(">=8.2")
+class TestLauncherTestSpecCrossVersionSpec extends TestLauncherSpec {
+
+    def setup() {
+        withFailingTest() // ensures that withTestsFor statements are not ignored
+    }
+
+    @TargetGradleVersion("=8.1")
```
Comment (bamboo) [inline subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r82/TestLauncherTestSpecCrossVersionSpec.groovy:37]: ❓ Isn't this method annotation in conflict with the class level annotation?

Comment (bamboo) [reply subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r82/TestLauncherTestSpecCrossVersionSpec.groovy:37]: Ok, conceptually though, isn't it odd to have a test targeted at `=8.1` in a class targeted at `>=8.2`?

Comment (donat) [reply subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r82/TestLauncherTestSpecCrossVersionSpec.groovy:37]: Correct, although I only wanted to prove that we had a broken behavior which we fixed in 8.2. I understand the confusion though.

Comment (donat) [reply subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r82/TestLauncherTestSpecCrossVersionSpec.groovy]: Fixed, ptal.

Comment (bamboo) [reply subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r82/TestLauncherTestSpecCrossVersionSpec.groovy]: I still see a test seemingly in conflict with the class level annotation but I don't have a feeling strong enough to object.

**Review by bamboo — APPROVED** [review]

File: subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r82/TestLauncherTestSpecCrossVersionSpec.groovy (hunk 1 above)
Comment (donat) [reply subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r82/TestLauncherTestSpecCrossVersionSpec.groovy]: This is a common pattern in cross-version tests. On the class level we define a range of target gradle versions (typically from current to all future versions) and override it on the method level for tests that describe old behavior.

**Comment by donat** [comment]
@bot-gradle test and merge

**Pull request merged**